dcc-repository-song/src/main/java/org/icgc/dcc/repository/song/core/SongProcessor.java

```diff
@@ -299,20 +300,20 @@ String getMetadataPath(SongAnalysis a) {
   }
 
   IndexFile getIndexFile(List<SongFile> files, String name) {
-    SongFile i = null;
+    Optional<SongFile> sf = Optional.empty();
     if (hasExtension(name, "BAM")) {
-      i = getSongIndexFile(files, name + ".BAI");
+      sf = getSongIndexFile(files, name + ".BAI");
     }
     if (hasExtension(name, "VCF")) {
-      i = getSongIndexFile(files, name + ".IDX");
-      if (i == null) {
-        i = getSongIndexFile(files, name + ".TCG");
-      }
+      sf = Stream.of(".TBI", ".IDX", ".TCG")
+              .map(suffix -> getSongIndexFile(files, name + suffix))
+              .filter(Optional::isPresent).findFirst().orElse(Optional.empty());
```

Comment:
you can also change 
```java
.filter(Optional::isPresent).findFirst().orElse(Optional.empty());
```
to be 
```java
.filter(Optional::isPresent)
.map(Optional::get)
.findFirst();
```

My personal preference is to make chained stream calls and optional calls stacked vertically, because it makes it easier to read.